algosdk/abi/interface.py

```diff
@@ -49,3 +49,23 @@ def undictify(d: dict) -> "Interface":
         method_list = [Method.undictify(method) for method in d["methods"]]
         desc = d["desc"] if "desc" in d else None
         return Interface(name=name, desc=desc, methods=method_list)
+
+    def get_method_by_name(self, name: str) -> Method:
+        methods_filtered = list(
```

Comment:
similar bias here.... general question, and not for this PR. Seems like there's a lot of overlap between contract and interface. Having a common abstract supertype might make sense.

Reply:
I think a change like this is a little outside of the scope of this PR, but it is something we should consider.

I definitely dislike how much code is repeated between the two methods though. Perhaps a short term solution is to define a `get_method_by_name(methods: list[Method], name: str) -> Method` helper function which implements the logic, then call into it from both `Interface.get_method_by_name` and `Contract.get_method_by_name`? I think all SDKs would benefit from this @barnjamin

Reply:
If itll be abstracted in another PR anyway, I'm inclined to leave it as is for now unless it makes you want to puke

Reply:
I've not thought through all the implications of making `Contract` inherit from `Interface`, so I'm not sure if/when we'd want to make that change. In an effort to not hold up these PRs because of that, I think they should be mergable on their own, which means trying to avoid duplicate code.

From a maintenance perspective there's way fewer opportunities for problems if the function is implemented in just one place.

Reply:
Makes sense, just added it